airbyte-cdk/bulk/toolkits/extract-cdc/src/main/kotlin/io/airbyte/cdk/read/cdc/CdcPartitionReader.kt

```diff
@@ -132,36 +131,32 @@ class CdcPartitionReader<T : Comparable<T>>(
         private val coroutineContext: CoroutineContext,
     ) : Consumer<ChangeEvent<String?, String?>> {
 
-        override fun accept(event: ChangeEvent<String?, String?>) {
+        override fun accept(changeEvent: ChangeEvent<String?, String?>) {
+            val event = DebeziumEvent(changeEvent)
             numEvents.incrementAndGet()
-            // Get SourceRecord object if possible.
-            // This object is the preferred way to obtain the current position.
-            val sourceRecord: SourceRecord? =
-                (event as? EmbeddedEngineChangeEvent<*, *, *>)?.sourceRecord()
-            if (sourceRecord == null) numEventsWithoutSourceRecord.incrementAndGet()
-            // Debezium outputs a tombstone event that has a value of null. This is an artifact
-            // of how it interacts with kafka. We want to ignore it. More on the tombstone:
-            // https://debezium.io/documentation/reference/stable/transformations/event-flattening.html
-            val debeziumRecordValue: DebeziumRecordValue? =
-                event.value()?.let { DebeziumRecordValue(Jsons.readTree(it)) }
+            if (event.sourceRecord == null) numEventsWithoutSourceRecord.incrementAndGet()
             // Process records, ignoring heartbeats which are only used for completion checks.
             val eventType: EventType = run {
-                if (debeziumRecordValue == null) return@run EventType.TOMBSTONE
-                if (debeziumRecordValue.isHeartbeat) return@run EventType.HEARTBEAT
-                val debeziumRecordKey = DebeziumRecordKey(Jsons.readTree(event.key()))
-                val deserializedRecord: DeserializedRecord =
-                    readerOps.deserialize(debeziumRecordKey, debeziumRecordValue)
-                        ?: return@run EventType.RECORD_DISCARDED_BY_DESERIALIZE
+                // Debezium outputs a tombstone event that has a value of null. This is an artifact
+                // of how it interacts with kafka. We want to ignore it. More on the tombstone:
+                // https://debezium.io/documentation/reference/stable/transformations/event-flattening.html
+                if (event.isTombstone) return@run EventType.TOMBSTONE
+                if (event.isHeartbeat) return@run EventType.HEARTBEAT
+                if (event.key == null || event.value == null) return@run EventType.INVALID_JSON
                 val streamRecordConsumer: StreamRecordConsumer =
-                    streamRecordConsumers[deserializedRecord.streamID]
+                    findStreamRecordConsumer(event.key, event.value)
                         ?: return@run EventType.RECORD_DISCARDED_BY_STREAM_ID
+                val deserializedRecord: DeserializedRecord =
+                    readerOps.deserialize(event.key, event.value, streamRecordConsumer.stream)
+                        ?: return@run EventType.RECORD_DISCARDED_BY_DESERIALIZE
```

Comment:
nothing here really changed except that:
- the `event.key` and `event.value` json-deserialization are wrapped in a `runCatching` to handle exceptions gracefully
- the stream ID gets validated before `deserialize` instead of after